synapse/storage/_base.py

```diff
@@ -113,9 +116,14 @@ def _attempt_to_invalidate_cache(
         try:
             cache = getattr(self, cache_name)
         except AttributeError:
-            # We probably haven't pulled in the cache in this worker,
-            # which is fine.
-            return
+            # Check if an externally defined module cache has been registered
+            external_func = self.external_cached_functions.get(cache_name)
+            if external_func:
+                cache = external_func
+            else:
+                # We probably haven't pulled in the cache in this worker,
+                # which is fine.
+                return False
```

Comment:
```suggestion
            cache = self.external_cached_functions.get(cache_name)
            if not cache:
                # We probably haven't pulled in the cache in this worker,
                # which is fine.
                return False
```

just came to mind, don't mind either way (not sure if mypy will even let you have this one)

Reply:
mypy was happy so let's go !